### PR TITLE
[janus-gateway] Dependencies cleanup + assorted tweaks

### PR DESCRIPTION
**Pull request by i8-pi**
Hello 👋. Thanks for the quick response on the AUR page on the package. I think this place is a more convenient for a discussion.
  
I still believe the split between make/optdepends I proposed is correct, because the janus binary does not directly link against anything in both optdepends and makedepends. If the features are enabled during configure, then a shared object gets built and installed under /usr/lib/janus (not scanned by ldconfig). The janus binary does not link against those either, but uses dlopen on them if they are not disabled in the config file. This means you can enable everything during configure but not have the dependencies installed at run time, and the binary will still run (with a few warnings, but can be avoided with the right config)

I have tested this with everything enabled during configure but nothing (except curl) in makedepends installed at runtime, and successfully ran the echo test and videoroom demos

Other changes

- Removed the HOME directory. It does not seem to be required.
- Removed cmake from makedepends. It's only used in the upstream travis setup
- Removed the javascript modules configure flag. It only adds a bundled version of the JS, and is IMO not useful for most people. Plus, tt wasn't enabling anything anyway since it had a typo

I was thinking of removing the hard coded janus UID as well and let systemd-sysusers allocate one on demand, but I'm worried it might break backwards compatibility

### CODE REVIEW

**Comment by alerque** [comment]
Thanks for the explanations and especially the testing you've done on this. I think the upstream project is being a bit unorthodox here, but it seems like you are correct about the way they handle runtime dependencies.

I'd love to see the hard coded UID go away too, but we should be careful about not breaking on upgrade and I don't have time to test that migration right now with both an upgrade and clean install.

Everything else looks great here. I'll merge as soon as I do a wee bit of testing myself.

File: janus-gateway/janus.conf [hunk 1]
```diff
@@ -1 +1 @@
-u janus 531 "Janus WebRTC gateway user" /var/lib/janus /bin/bash
+u janus 531 "Janus WebRTC gateway user"
```
Comment (alerque) [inline janus-gateway/janus.conf:1]: Making the home directory go away if it is actually unused is fine, but should we also assign a dummy shell like `/bin/true`? Or in the brave new world of systemd managed user accounts is that not helful?

**Comment by alerque** [comment]
I added you as a co-maintainer to the AUR repo so if need arises and I'm out of touch you could actually push directly too, but feel free to use PRs here on this repo as a staging ground and more robust review mechanism than AUR provides.

Thanks again for your work on this package.

File: janus-gateway/janus.conf (hunk 1 above)
Comment (i8-pi) [reply janus-gateway/janus.conf:1]: Turns out, you end up with /usr/bin/nologin if you leave it empty here. I guess you can trust the defaults in this brave new world 😄

**Comment by i8-pi** [comment]
@alerque Hi, responding to your question on the janus-gateway AUR package page on why the optional dependencies are there, my first message in this thread should answer your questions

Also, good luck on the TU application 👍

**Comment by alerque** [comment]
@i8-pi Thanks for refreshing my memory!

**Pull request merged**